### PR TITLE
Upgrade Newtonsoft.Json dependency

### PR DESCRIPTION
**Pull request by xavierdecoster**
Fixes https://github.com/NuGet/Engineering/issues/2508

### CODE REVIEW

File: src/NuGet.Services.KeyVault/NuGet.Services.KeyVault.csproj [hunk 1]
```diff
@@ -75,12 +75,18 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
+    <PackageReference Include="Microsoft.Azure.Common">
+      <Version>2.2.1</Version>
```
Comment (xavierdecoster) [inline src/NuGet.Services.KeyVault/NuGet.Services.KeyVault.csproj]: This upgrade isn't strictly required, but it's the latest available dependency which seems to clean-up its own transitive dependencies a lot.

File: src/NuGet.Services.KeyVault/NuGet.Services.KeyVault.csproj [hunk 2]
```diff
@@ -75,12 +75,18 @@
...
+    </PackageReference>
     <PackageReference Include="Microsoft.Azure.KeyVault">
       <Version>1.0.0</Version>
     </PackageReference>
     <PackageReference Include="Microsoft.IdentityModel.Clients.ActiveDirectory">
       <Version>3.19.4</Version>
     </PackageReference>
+    <PackageReference Include="Newtonsoft.Json">
+      <Version>8.0.1</Version>
```
Comment (xavierdecoster) [inline src/NuGet.Services.KeyVault/NuGet.Services.KeyVault.csproj]: CG guidance states to upgrade to v8 minimum

**Comment by xavierdecoster** [comment]
Confirmed in CG that this change fixes the issue.

File: src/NuGet.Services.KeyVault/NuGet.Services.KeyVault.csproj (hunk 1 above)
Comment (joelverhagen) [reply src/NuGet.Services.KeyVault/NuGet.Services.KeyVault.csproj]: How does this change the graph? If this is not necessary can we split it into its own PR?

File: src/NuGet.Services.KeyVault/NuGet.Services.KeyVault.csproj (hunk 2 above)
Comment (joelverhagen) [reply src/NuGet.Services.KeyVault/NuGet.Services.KeyVault.csproj]: Can we do `10.0.2`? It's more consistent with the rest of the repo:
https://github.com/search?utf8=%E2%9C%93&q=Newtonsoft.Json+repo%3ANuGet%2FServerCommon+extension%3Acsproj&type=Code&ref=advsearch&l=&l=

File: src/NuGet.Services.KeyVault/NuGet.Services.KeyVault.csproj (hunk 1 above)
Comment (xavierdecoster) [reply src/NuGet.Services.KeyVault/NuGet.Services.KeyVault.csproj]: Uninstalling this 2.2.1 update (and reverting to the transient dependency dictated by `Microsoft.Azure.KeyVault` version `1.0.0`; which is `Microsoft.Azure.Common (>= 2.0.4 && < 3.0.0)`, results in these graph changes:

image

I'll revert this particular change for now to be excluded from this PR.

File: src/NuGet.Services.KeyVault/NuGet.Services.KeyVault.csproj (hunk 2 above)
Comment (xavierdecoster) [reply src/NuGet.Services.KeyVault/NuGet.Services.KeyVault.csproj]: Yes we can!

Comment (xavierdecoster) [reply src/NuGet.Services.KeyVault/NuGet.Services.KeyVault.csproj]: Fixed in https://github.com/NuGet/ServerCommon/pull/311/commits/42226d732b58307ee6f88483d5dbd524a4ae687d

File: src/NuGet.Services.KeyVault/NuGet.Services.KeyVault.csproj (hunk 1 above)
Comment (xavierdecoster) [reply src/NuGet.Services.KeyVault/NuGet.Services.KeyVault.csproj]: Reverted in https://github.com/NuGet/ServerCommon/pull/311/commits/42226d732b58307ee6f88483d5dbd524a4ae687d

**Pull request merged**